taxcalc/functions.py

```diff
@@ -1604,60 +1583,36 @@ def C1040(e07400, e07180, e07200, c07220, c07230, e07250,
 
 
 @iterate_jit(nopython=True)
-def DEITC(c08795, c59660, c09200, c07100, c08800, c05800, _othertax):
+def DEITC(c08795, c59660, c09200, c07100, c08800, c05800, _avail, _othertax):
 
     # Decomposition of EITC
-
-    if c08795 > 0 and c59660 > 0 and c08795 <= c59660:
-        c59680 = c08795
-
-        _comb = c59660 - c59680
-
-    else:
-
-        c59680 = 0.
-        _comb = 0.
-
-    if c08795 > 0 and c59660 > 0 and c08795 > c59660:
-        c59680 = c59660
+    _comb = 0
+    c10950 = 0
+    c59680 = min(c59660, max(0, _avail))
+    _avail = _avail - c59680
+    _avail = _avail + _othertax
+    c59700 = min(_avail, c59660 - c59680)
+    c59720 = c59660 - c59680 - c59700
```

Comment:
There shouldn't be any iteration. The SOI has a definitive way to stack 
credits, and taxcalc.sas follows it, or at least the version I have now 
does. There have been some slight problems in the September version, and 
bigger problems in earlier versions.

dan

On Mon, 5 Oct 2015, Sean.Wang wrote:

> 
> In taxcalc/functions.py:
> 
> > -        _comb = c59660 - c59680
> > -
> > -    else:
> > -
> > -        c59680 = 0.
> > -        _comb = 0.
> > -
> > -    if c08795 > 0 and c59660 > 0 and c08795 > c59660:
> > -        c59680 = c59660
> > +    _comb = 0
> > +    c10950 = 0
> > +    c59680 = min(c59660, max(0, _avail))
> > +    _avail = _avail - c59680
> > +    _avail = _avail + _othertax
> > +    c59700 = min(_avail, c59660 - c59680)
> > +    c59720 = c59660 - c59680 - c59700
> 
> A little bit confused here, but I think here we are using allocate credits to get the
> EIC refundable portion iteratively. The update makes more sense to me in the way that
> it involves extra intermediate steps of iteration.
> 
> —
> Reply to this email directly or view it on
> GitHub.[AHvQVXKw7kBkM238ifPVcKCwYkEor3ytks5o4uoRgaJpZM4GHeOp.gif]
> 
> 
>